internal/services/graph/graph_account_resource_test.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type AccountTestResource struct{}
+
+func TestAccGraphAccount_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccGraphAccount_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccGraphAccount_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccGraphAccount_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r AccountTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := graphservicesprods.ParseAccountID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Graph.V20230413.Graphservicesprods.AccountsGet(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r AccountTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_graph_account" "test" {
+  name                = "acctesta-%[2]d"
+  app_id              = azuread_application.test.application_id
```

Comment:
this resource is depend on an `azuread_application`, however through my local test the resource creation may fail occationally with error even after the ad app is created: `performing AccountsCreateAndUpdate: unexpected status 400 with error: BadRequest: The application was not found in this tenant.`. The service team told that this is hard to change and advices to use a predefined azure ad app, can we do this?